frontend/src/functions/filteredArrayByKeyword.js

```diff
@@ -0,0 +1,19 @@
+const textIsIncludesKeyword = require("./textIsIncludesKeyword");
+
+const filteredArrayByKeyword = (objectArray, keyword) => {
+  return keyword.length === 0
+    ? objectArray
+    : objectArray.filter(
+        (object) =>
+          textIsIncludesKeyword(object.name, keyword) ||
+          object.products.some((product) =>
+            textIsIncludesKeyword(product, keyword)
+          ) ||
+          textIsIncludesKeyword(object.intro, keyword) ||
+          textIsIncludesKeyword(object.introductionLong, keyword)
```

Comment:
És azt megnézted, hgy mi történik, ha nincs introduction például? Lemegy egy null a textIsIncludesKeyword első paraméterének, ami leküldi a matchFormatnak, ami ráhív egy normalize-t: TypeError: Cannot read properties of null (reading 'normalize')
Erre megint csak egy elég jó tesztesetet lehet írni...

Reply:
Elvileg backend felől le van kezelve, hogy ne is kerüljön be az adatbázisba ha ezek az adattagok null-ok, de mindjárt nézem.

Reply:
Lehet null, legalábbis a default betöltött demóadatoknál null lesz :)

Reply:
Akkor viszont intézem :D